### PR TITLE
Re-order the tap_common/init.d/rcS script contents to better match the px4_common/init.d/rcS script.

### PR DESCRIPTION
**Pull request by mcsauder**
Hi there,

I noticed that the `tap_common/init.d/rcS` script very closely resembles the `px4_common/init.d/rcS` script and makes calls to files in the `px4_common/init.d/` directory.

Given that, it seemed appropriate to reduce/re-use code in `rc.vehicle_setup` and also structure the script in the same basic order/outline as the `px4_common/init.d/rcS` file.  This PR is that work.

**In Full Disclosure**... I am unaware if CI tests are in place to test this PR and I do not have hardware to test this PR myself.  This PR requires testing/validating if any help might be available... Thank you in advance!

Please let me know if you have any questions on this PR!

-Mark

### CODE REVIEW

**Comment by potaito** [comment]
@mcsauder thanks, that was on my TODO list. I'll have a look and test with hardware.

**Comment by mcsauder** [comment]
Hi @potaito , thank you for looking at this PR.

I think I understand it a bit better than yesterday.  I think it was an unintended omission in the current master branch `tap_common/init.d/rcS`, rather than a reference to the `px4fmu_common/init.d/rc.ugv_apps` file... It looks like because that logic block was never entered, it never looked for a `rc.ugv_apps` file in the same directory and that is how the omission went undetected...  I added a commit that copies those files into the `tap_common/init.d/` directory. Would you be willing to let me know if this is needed?

Also, is there a better manner to handle this other than copy/paste into the

**Comment by mcsauder** [comment]
Thanks @mhkabir! 

OK, so should the rcS script be let the way it is with the UGV code removed? (And of course remove rc.vehicle_setup and rc.ugv_apps and rc.ugv_defaults?)

**Comment by mcsauder** [comment]
Alright!  I've reverted a few things, rebased the commits, and tidied up.

Please have a look and let me know your thoughts!

**Comment by mhkabir** [comment]
Looks good to me. @potaito Can you quickly verify?

**Comment by potaito** [comment]
Unfortunately I am unable to test it since our internal version is yet again different. Too different to just cherry-pick and spin it up for a test. Guess this can only be tested with the Intel Aero for which is it intended...

My mid-term goal was to merge the entire `ROMFS/tap_common` with `ROMFS/px4fmu_common`. Not sure if that is even possible...

**Comment by mcsauder** [comment]
Thanks @potaito , 

> When you modified the order with which the different modules are launched, what reference did you use? Your changes don't strictly resemble ROMFS/px4fmu_common/init.d/rcS or do they?

Agreed. This refactor doesn't 'strictly' resemble `ROMFS/px4fmu_common/init.d/rcs`, its just a first pass at re-aligning the two to keep more of the files in roughly the same order.  At best, they would use the same file, but this PR at least attempts to keep them better aligned.

File: ROMFS/tap_common/init.d/rcS [hunk 1]
```diff
@@ -130,7 +138,6 @@ then
 	param set SYS_AUTOCONFIG 0
 	param save
 fi
-unset AUTOCNF
```
[the remark this answers is not in the document]
Comment (mcsauder) [reply ROMFS/tap_common/init.d/rcS]: Good catch. I should have moved it to the top/bottom of the script.  That is done and pushed up!  Thank you for catching that!

File: ROMFS/tap_common/init.d/rcS [hunk 2]
```diff
@@ -145,97 +151,58 @@ then
...
-
-# Start MAVLink on the gimbal port
-#mavlink start -r 1200 -d /dev/ttyS1
-
-# Start MAVLink on USB, developers can use the MAVLink shell
-mavlink start -r 60000 -d /dev/ttyACM0 -m config
-
 #
-# Logging
+# Start the RC input driver
```
Comment (bkueng) [inline ROMFS/tap_common/init.d/rcS]: fmu is not used for RC input anymore.
There should be a separate `rc_input start`

**Comment by potaito** [comment]
@bkueng 
> Intel Aero uses the px4fmu_common ROMFS, so you can have full authority over tap_common :)

Okay hold on, that means currently nobody is using these files, at all. :crying_cat_face: 
What we have in the fork is again slightly different, so I can't cherrypick this PR (already tried). I'm inclined to say that we can delete the entire `ROMFS/tap_common` and add the new files back once needed. But I don't want to make this decision alone. @simonegu can you comment?

File: ROMFS/tap_common/init.d/rcS (hunk 2 above)
Comment (mcsauder) [reply ROMFS/tap_common/init.d/rcS]: Done. Thanks @bkueng .

**Comment by mcsauder** [comment]
Hi @potaito and @simonegu , 

> Okay hold on, that means currently nobody is using these files, at all.

Should a PR deprecating this directory be submitted and I'll close this PR accordingly?

**Comment by simonegu** [comment]
@potaito @mcsauder we can probably deprecate it, we would also need to get rid of the `tap-v1_default` target which is using this.

**Comment by mcsauder** [comment]
Thanks @simonegu.

I've opened up PR#10629 to advance the discussion in this PR.  Let me know if this was the correct direction to move!

**Comment by mcsauder** [comment]
Closing in favor of PR #10629.  (If #10629 is unsuccessful we can revisit this PR.)

**Pull request closed without merging**